New package: py-deeplabcut

qwertos opened the pull request.

qwertos committed:
[py-deeplabcut] created template

qwertos committed:
[py-deeplabcut] added dependencies

qwertos committed:
[py-deeplabcut] widening support for numpy

qwertos committed:
[py-deeplabcut] maybe equivalent? we'll see

qwertos committed:
[py-deeplabcut] added homepage and description. removed fixmes

qwertos committed:
[py-deeplabcut] added most recent stable version

qwertos committed:
[py-deeplabcut] removed beta version

adamjstewart commented on var/spack/repos/builtin/packages/py-deeplabcut/package.py:
```diff
@@ -0,0 +1,55 @@
+    depends_on("py-numba@0.51.1", type=('build', 'run'))
+    depends_on("py-matplotlib@3.1.3", type=('build', 'run'))
+    depends_on("py-moviepy@:1.0.1", type=('build', 'run'))
+    depends_on("py-numpy@1.16.4:1.999", type=('build', 'run'))
```
```suggestion
    depends_on("py-numpy@1.16.4", type=('build', 'run'))
```

adamjstewart commented on var/spack/repos/builtin/packages/py-deeplabcut/package.py:
```diff
@@ -0,0 +1,55 @@
+    depends_on("py-pyyaml", type=('build', 'run'))
+    depends_on("py-requests", type=('build', 'run'))
+    depends_on("py-ruamel-yaml@0.15.0:", type=('build', 'run'))
+    depends_on("py-setuptools", type=('build', 'run'))
```
You can remove this above so it's not duplicated

qwertos committed:
[py-deeplabcut] removed duplicate setuptools

qwertos committed:
[py-deeplabcut] restrcting py-numpy back to specification in setup.py

Restricting py-pandas to the version that works for the restricted
version of py-numpy to help the concretizer.

alalazo commented:
@qwertos Can you please rebase and ping me and @adamjstewart back?

qwertos commented:
@alalazo @adamjstewart rebased and pinged

adamjstewart commented on var/spack/repos/builtin/packages/py-deeplabcut/package.py:
```diff
@@ -0,0 +1,54 @@
+    depends_on("py-moviepy@:1.0.1", type=('build', 'run'))
+    depends_on("py-numpy@1.16.4", type=('build', 'run'))
+    # depends_on("py-opencv-python-headless", type=('build', 'run'))
+    depends_on("opencv+python", type=('build', 'run'))
```
I changed the `opencv` package so now there is no `+python` variant. As far as I can tell, `deeplabcut` only supports Python 3? So I would change this:
```suggestion
    depends_on("opencv+python3", type=('build', 'run'))
```

adamjstewart commented on var/spack/repos/builtin/packages/py-deeplabcut/package.py:
```diff
@@ -0,0 +1,54 @@
+    depends_on("py-numpy@1.16.4", type=('build', 'run'))
+    # depends_on("py-opencv-python-headless", type=('build', 'run'))
+    depends_on("opencv+python", type=('build', 'run'))
+    depends_on("py-pandas@1.0.1:1.1.999", type=('build', 'run'))
```
Where does this upper bound come from?

adamjstewart commented on var/spack/repos/builtin/packages/py-deeplabcut/package.py:
```diff
@@ -0,0 +1,54 @@
+
+    version('2.1.10', sha256='457ae694ec91d4903d7077c4ef14a00444a9f8932f3d1cbe08d1c2d58c326df5')
+
+    depends_on('py-setuptools', type='build')
```
Can you check if this is needed at run-time?

adamjstewart commented on var/spack/repos/builtin/packages/py-deeplabcut/package.py:
```diff
@@ -0,0 +1,54 @@
+    depends_on("py-certifi", type=('build', 'run'))
+    depends_on("py-chardet", type=('build', 'run'))
+    depends_on("py-click", type=('build', 'run'))
+    depends_on("py-cython", type=('build', 'run'))
```
Can you check if this is needed at run-time?

alalazo commented:
@qwertos

alalazo commented:
Closing this PR as stale, but feel free to reopen if you want to continue working on it.
